tutorials/super_blink/super_blink.c

```diff
@@ -0,0 +1,51 @@
+#include "blink_config.h"
+
+#include <avr/interrupt.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#define HW_TIMER_Hz     (30)
+#define SW_TIMER_SCALER (6)
+
+static volatile bool run_blink_led0 = true;
+static volatile bool run_blink_led1 = true;
+uint8_t counter = 0;
+
+void timer0_callback(void) {
+    run_blink_led0 = true;
+}
+
+void sw_timer_callback(void) {
```

Comment:
this does work, but in my personal opinion this pattern is a little over complicated, ex. there's not necessarily a need for a separate callback function like there is in a hardware CTC match, so it'd probably be a little bit easier to set `run_blink_led1` in the `sw_timer` function.

if you really wanted to trim it down, one way to do this would be to define a macro equal to HW_TIMER_Hz over your scaler (which I believe would be six) and increment your counter in the run_blink_led0 block, so you'd have something like

```
#define 5_HZ_SCALER 6
...
if  (run_blink_led0) {
    counter++
    ...
}

if  (counter % 5_HZ_SCALER == 0) {
   ...
}
```
which would also allow you to locally scope the counter to the main function instead of having it as global. not necessarily right or wrong, just how I would personally write it